### PR TITLE
Layer API: change exceptions to warnings

### PR DESCRIPTION
**Pull request by moving-bits**

### CODE REVIEW

File: mapsforge-map/src/main/java/org/mapsforge/map/layer/Layer.java [hunk 1]
```diff
@@ -23,11 +23,15 @@
...
 import org.mapsforge.map.model.DisplayModel;
 
+import java.util.logging.Logger;
+
 public abstract class Layer {
     protected DisplayModel displayModel;
     private Redrawer assignedRedrawer;
     private boolean visible = true;
 
+    private static final Logger LOGGER = Logger.getLogger(Layers.class.getName());
```
Comment (devemux86) [inline mapsforge-map/src/main/java/org/mapsforge/map/layer/Layer.java:33]: We should use `Layer.class.getName` here.

File: mapsforge-map/src/main/java/org/mapsforge/map/layer/Layer.java [hunk 2]
```diff
@@ -169,7 +173,7 @@ protected void onRemove() {
 
     final synchronized void assign(Redrawer redrawer) {
         if (this.assignedRedrawer != null) {
-            throw new IllegalStateException("layer already assigned");
+            LOGGER.warning("layer already assigned");
```
Comment (devemux86) [inline mapsforge-map/src/main/java/org/mapsforge/map/layer/Layer.java:176]: Probably it is better to `return;` after the warning?

File: mapsforge-map/src/main/java/org/mapsforge/map/layer/Layer.java [hunk 3]
```diff
@@ -178,7 +182,7 @@ final synchronized void assign(Redrawer redrawer) {
 
     final synchronized void unassign() {
         if (this.assignedRedrawer == null) {
-            throw new IllegalStateException("layer is not assigned");
+            LOGGER.warning("layer is not assigned");
```
Comment (devemux86) [inline mapsforge-map/src/main/java/org/mapsforge/map/layer/Layer.java:185]: Same here, probably it is better to `return;` after the warning?

**Comment by devemux86** [comment]
@moving-bits thanks! Please see my comments.

By the way your fork is not rebased on the latest `master` branch?
The checks have passed in this pull request, but it could cause conflicts in other ones. :slightly_smiling_face:

File: mapsforge-map/src/main/java/org/mapsforge/map/layer/Layer.java (hunk 1 above)
Comment (moving-bits) [reply mapsforge-map/src/main/java/org/mapsforge/map/layer/Layer.java]: c&p error, fixed

File: mapsforge-map/src/main/java/org/mapsforge/map/layer/Layer.java (hunk 2 above)
Comment (moving-bits) [reply mapsforge-map/src/main/java/org/mapsforge/map/layer/Layer.java:176]: wasn't sure about what is more important - `onAdd()` getting called or directly exiting on invalid state - but added `return` now

File: mapsforge-map/src/main/java/org/mapsforge/map/layer/Layer.java (hunk 3 above)
Comment (moving-bits) [reply mapsforge-map/src/main/java/org/mapsforge/map/layer/Layer.java:186]: similar to above

**Comment by devemux86** [comment]
@moving-bits thanks!

**Pull request merged**